From e32cff16c537898e8db81a793cc2335bed5c35f7 Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Fri, 13 Mar 2015 08:45:02 +0100
Subject: [PATCH 0512/1800] EMsgComposer - Avoid runtime warning when opening a
 template that was not written in WebKit based composer

This change will also as a consequence show a dialog whether the user
wants to lose formatting.
---
 composer/e-msg-composer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/composer/e-msg-composer.c b/composer/e-msg-composer.c
index 49424d9ccf..560690b9f6 100644
--- a/composer/e-msg-composer.c
+++ b/composer/e-msg-composer.c
@@ -3594,7 +3594,7 @@ e_msg_composer_new_with_message (EShell *shell,
 		flags = g_strsplit (format, ", ", 0);
 		for (i = 0; flags[i]; i++) {
 			if (g_ascii_strcasecmp (flags[i], "text/html") == 0) {
-				if (g_ascii_strcasecmp (composer_mode, "text/html") == 0) {
+				if (composer_mode && g_ascii_strcasecmp (composer_mode, "text/html") == 0) {
 					e_html_editor_view_set_html_mode (
 						view, TRUE);
 				} else {
@@ -3602,7 +3602,7 @@ e_msg_composer_new_with_message (EShell *shell,
 						view, FALSE);
 				}
 			} else if (g_ascii_strcasecmp (flags[i], "text/plain") == 0) {
-				if (g_ascii_strcasecmp (composer_mode, "text/html") == 0) {
+				if (composer_mode && g_ascii_strcasecmp (composer_mode, "text/html") == 0) {
 					e_html_editor_view_set_html_mode (
 						view, TRUE);
 				} else {
-- 
2.15.0

